packages/block-library/src/gallery/utils.js

```diff
@@ -22,8 +22,17 @@ import {
  */
 export function getHrefAndDestination( image, destination ) {
 	// Need to determine the URL that the selected destination maps to.
-	// Gutenberg and WordPress use different constants so the new link
-	// destination also needs to be tweaked.
+	// Gutenberg and WordPress use different constants so if image_default_link_type
+	// option is set we need to map file -> media and post -> attachment
+	switch ( destination ) {
```

Comment:
Do we need to add the additional `switch` statement?

Could we add these cases to the below switch statement instead, something like:

```
	switch ( destination ) {
		case LINK_DESTINATION_MEDIA:
		case 'file':
			return {
				href: image?.source_url || image?.url, // eslint-disable-line camelcase
				linkDestination: IMAGE_LINK_DESTINATION_MEDIA,
			};
		case LINK_DESTINATION_ATTACHMENT:
		case 'post':
			return {
				href: image?.link,
				linkDestination: IMAGE_LINK_DESTINATION_ATTACHMENT,
			};
```

I was also wondering if it's worth creating another couple of constants in `constants.js` so that these two new ones are centralised? (Though I'm not sure if we'll need to use them elsewhere 🤔)

Reply:
Yeh, that tidies it up nicely - have pushed that suggestion.

Reply:
Perfect, that looks good to me!